dot/rpc/subscription/subscription.go

```diff
@@ -6,54 +6,70 @@ import (
 	"strconv"
 )
 
-var errUknownParamSubscribeID = errors.New("invalid params format type")
-var errCannotParseID = errors.New("could not parse param id")
-var errCannotFindListener = errors.New("could not find listener")
-var errCannotFindUnsubsriber = errors.New("could not find unsubsriber function")
+type methodName string
+
+const (
+	authorSubmitAndWatchExtrinsic  methodName = "author_submitAndWatchExtrinsic" //nolint
+	chainSubscribeNewHeads                    = "chain_subscribeNewHeads"
+	chainSubscribeNewHead                     = "chain_subscribeNewHead"
+	chainSubscribeFinalizedHeads              = "chain_subscribeFinalizedHeads"
+	stateSubscribeStorage                     = "state_subscribeStorage"
+	stateSubscribeRuntimeVersion              = "state_subscribeRuntimeVersion"
+	grandpaSubscribeJustifications            = "grandpa_subscribeJustifications"
+)
 
-type unsubListener func(reqid float64, l Listener, params interface{})
 type setupListener func(reqid float64, params interface{}) (Listener, error)
 
+var (
+	errUknownParamSubscribeID = errors.New("invalid params format type")
+	errCannotParseID          = errors.New("could not parse param id")
+	errCannotFindListener     = errors.New("could not find listener")
+	errCannotFindUnsubsriber  = errors.New("could not find unsubsriber function")
+)
+
 func (c *WSConn) getSetupListener(method string) setupListener {
-	switch method {
-	case "chain_subscribeNewHeads", "chain_subscribeNewHead":
+	switch methodName(method) {
+	case authorSubmitAndWatchExtrinsic:
+		return c.initExtrinsicWatch
+	case chainSubscribeNewHeads, chainSubscribeNewHead:
 		return c.initBlockListener
-	case "state_subscribeStorage":
+	case stateSubscribeStorage:
 		return c.initStorageChangeListener
-	case "chain_subscribeFinalizedHeads":
+	case chainSubscribeFinalizedHeads:
 		return c.initBlockFinalizedListener
-	case "state_subscribeRuntimeVersion":
+	case stateSubscribeRuntimeVersion:
 		return c.initRuntimeVersionListener
-	case "grandpa_subscribeJustifications":
+	case grandpaSubscribeJustifications:
 		return c.initGrandpaJustificationListener
+	default:
+		// in case the method is unsafe search for it in another function
+		if c.UnsafeEnabled {
+			return c.getSetupUnsafeListener(method)
+		}
+
+		return nil
+	}
+}
+
+func (c *WSConn) getSetupUnsafeListener(method string) setupListener {
```

Comment:
Is this function need? This is just returning `nil`

Reply:
removed